crates/cubecl-linalg/src/matmul/kernels/naive.rs

```diff
@@ -54,13 +54,13 @@ fn matmul_kernel<N: Numeric>(
     k /= line_size.runtime();
 
     for i in 0..k {
-        let lhs_index = row * k + i + offset_lhs;
-        let rhs_index = col * k + i + offset_rhs;
+        let lhs_index = row * lhs.stride(rank - 2) / line_size + i + offset_lhs;
+        let rhs_index = col * rhs.stride(rank - 2) / line_size + i + offset_rhs;
 
         sum += lhs[lhs_index] * rhs[rhs_index];
     }
 
-    let mut out_index = row * n_cols + col;
+    let mut out_index = row * out.stride(rank - 2) + col;
     out_index += offset_out;
```

Comment:
Are you sure you didn't change the correctness here?

Reply:
I increased the correctness, because it now also works with non-contiguous out tensors. `n_cols` is just the stride assuming contiguous memory, so `stride(rank - 2)` is strictly more correct.